ecl/hql/hqlcache.cpp

```diff
@@ -340,6 +342,161 @@ void convertSelectsToPath(StringBuffer & filename, const char * eclPath)
+{
+public:
+    HqlSimplifiedRecordTransformer() : QuickHqlTransformer(HqlSimplifiedRecordTransformerInfo, nullptr)
+    {
+    }
+
+    virtual IHqlExpression * createTransformed(IHqlExpression * expr)
+    {
+        IHqlExpression * body = expr->queryBody(false);
+        if (expr != body)
+            return transform(body);
+
+        node_operator op = expr->getOperator();
+        switch (op)
+        {
+        case no_record:
+            {
+                HqlExprArray children;
+                bool same = true;
+                ForEachChild(idx, expr)
+                {
+                    IHqlExpression * cur = expr->queryChild(idx);
+                    if (!cur->isAttribute())
+                    {
+                        OwnedHqlExpr mapped = transform(cur);
+                        if (!mapped)
+                            throw MakeStringExceptionDirect(0, "Cannot simplify record");
+                        if (mapped != cur)
+                            same = false;
+                        children.append(*(mapped.getClear()));
+
+                    }
+                    else
+                        same = false;
+                }
+                if (same)
+                    return LINK(expr);
+                return expr->clone(children);
+            }
+        case no_field:
+            {
+                ITypeInfo * type = expr->queryType();
+                if (type->getTypeCode()==type_alien)
+                    throw MakeStringExceptionDirect(0, "Cannot simplify record");
+                HqlExprArray children;
+                bool same = true;
+                ForEachChild(idx, expr)
+                {
+                    IHqlExpression * cur = expr->queryChild(idx);
+                    if (cur->isAttribute())
+                    {
+                        if (cur->queryName()==_implicitFpos_Atom)
+                            same = false;
+                        else
+                        {
+                            HqlExprArray attrs;
+                            bool same2 = true;
+                            ForEachChild(idx2, cur)
+                            {
+                                IHqlExpression * attr = cur->queryChild(idx2)->queryBody();
+                                if (!attr->isConstant())
+                                    throw MakeStringExceptionDirect(0, "Cannot simplify record");
+                                if (attr->isAttribute())
+                                {
+                                    attrs.append(*(LINK(attr)));
+                                }
+                                else
+                                {
+                                    IValue * val = attr->queryValue();
+                                    if (!val)
+                                        throw MakeStringExceptionDirect(0, "Cannot simplify record");
+                                    if (val->getTypeCode()==type_int)
```

Comment:
Not clear what this is doing either - this function could do with some comments to explain the logic.

Reply:
@ghalliday Comment added above to explain this line.